document/src/main/java/com.google.android.fhir.document/interfaces/SHLGenerator.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.fhir.document.interfaces
+
+import android.content.Context
+import android.widget.ImageView
+import com.google.android.fhir.document.dataClasses.SHLData
+import kotlinx.coroutines.CoroutineScope
+
+interface SHLGenerator {
```

Comment:
Let's split the SHL generation process into two steps, based on the code you jave:
* SHLink generation
* QR generation

Let's have two interfaces each representing each step"
```suggestion
interface SHLinkGenerator {
```

and

```
interface QRGenerator {
```

can you also change the folder strucutre to reflect functionality as opposed to having dataClasses/interfaces/utils. I would just do:

document/src/main/java/com.google.android.fhir.document/generate

and have all the code in this folder